src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -164,6 +165,18 @@ export class DatasetService {
     });
   }
 
+  public refreshCache(): void {
```

Comment:
nit: is this the right name? not sure what refreshing means...

Reply:
+1 ideally this should be renamed or this function should refetching the children for the current data structure.